UnitTests/TestWallet.cpp

```diff
@@ -1743,18 +1743,141 @@ TEST_F(TestWallet, MultipleKeys)
-   EXPECT_EQ(wallet->encryptionKeys()[1], pd2.metaData.encKey);
+   ASSERT_EQ(wallet->encryptionKeys().size(), 2);
+   ASSERT_EQ(wallet->encryptionTypes().size(), 2);
+   ASSERT_EQ(wallet->encryptionRank().n, 2);
+   ASSERT_EQ(wallet->encryptionKeys()[1], pd2.metaData.encKey);
+}
+
+TEST_F(TestWallet, RemoveAuthEid)
+{
+   const auto authEidKey1 = CryptoPRNG::generateRandom(32);
+   const auto authEidKey2 = CryptoPRNG::generateRandom(32);
+   const auto authEidKey3 = CryptoPRNG::generateRandom(32);
+   const bs::core::wallet::Seed seed{ SecureBinaryData::fromString("test seed"), NetworkType::TestNet };
+   const bs::wallet::PasswordData pd1{ authEidKey1, { bs::wallet::EncryptionType::Auth, BinaryData::fromString("test@example.com:1NhzmQEQOyIx/DHqRj7zULPomE65/97Bjv0AlykbEzVV:Phone1") } };
+   const bs::wallet::PasswordData pd2{ authEidKey2, { bs::wallet::EncryptionType::Auth, BinaryData::fromString("test@example.com:KJd8AxATUCQdCOZLEmYz3Y1PllRY4XxnD+LWxsSaOauv:Phone2") } };
+   const bs::wallet::PasswordData pd3{ authEidKey3, { bs::wallet::EncryptionType::Auth, BinaryData::fromString("test@example.com:KCj08HSOR5rooqTGjoJbISun9McTeLNfFGv5tMPKPjJW:Phone3") } };
+   auto wallet = std::make_shared<bs::core::hd::Wallet>("test1", "", seed, pd1, "./homedir"
+      , StaticLogger::loggerPtr);
+   ASSERT_NE(wallet, nullptr);
+
+   SecureBinaryData privKey;
+   {
+      bs::core::WalletPasswordScoped lock(wallet, pd1.password);
+      ASSERT_NO_THROW(wallet->getDecryptedRootXpriv());
+      privKey = wallet->getDecryptedRootXpriv();
+      ASSERT_FALSE(privKey.empty());
+
+      bool result = wallet->addPassword(pd2);
+      ASSERT_TRUE(result);
+
+      result = wallet->addPassword(pd3);
+      ASSERT_TRUE(result);
+   }
+
+   {
+      bs::core::WalletPasswordScoped lock(wallet, pd1.password);
+      ASSERT_NO_THROW(wallet->getDecryptedRootXpriv());
+      ASSERT_EQ(wallet->getDecryptedRootXpriv(), privKey);
+   }
+
+   {
+      bs::core::WalletPasswordScoped lock(wallet, pd2.password);
+      ASSERT_NO_THROW(wallet->getDecryptedRootXpriv());
+      ASSERT_EQ(wallet->getDecryptedRootXpriv(), privKey);
+   }
+
+   {
+      bs::core::WalletPasswordScoped lock(wallet, pd3.password);
+      ASSERT_NO_THROW(wallet->getDecryptedRootXpriv());
+      ASSERT_EQ(wallet->getDecryptedRootXpriv(), privKey);
+   }
+
+   ASSERT_EQ(wallet->encryptionKeys().size(), 3);
+   ASSERT_EQ(wallet->encryptionTypes().size(), 1);
+   ASSERT_EQ(wallet->encryptionRank().m, 1);
+   ASSERT_EQ(wallet->encryptionRank().n, 3);
+   ASSERT_EQ(wallet->encryptionKeys()[0], pd1.metaData.encKey);
+   ASSERT_EQ(wallet->encryptionKeys()[1], pd2.metaData.encKey);
+   ASSERT_EQ(wallet->encryptionKeys()[2], pd3.metaData.encKey);
+
+   {
+      // Remove one newly added password
+      bs::core::WalletPasswordScoped lock(wallet, pd1.password);
+      // FIXME: This will remove pd1 but we need to remove pd3 here
+      bool result = wallet->removePassword(pd3.metaData);
```

Comment:
You should check that the wallet isn't locked before attempting to remove the password. The password removal method on the armory side will needs to lock the wallet on its own, because the password used to decrypt keys within that lock lifetime is the one that will be removed. 

The locking mechanism in the terminal is contravention of armory's design as it keeps passwords in ram instead of actually prompting the user, and they precede the lock. The lock should precede the password.

In the current terminal password handling implementation the arg for removePassword is useless.